### PR TITLE
doc: Add release notes for 22570 (ignore banlist.dat)

### PR DESCRIPTION
**Pull request by maflcko**
Follow-up to #22570

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -66,6 +66,13 @@ New RPCs
 Build System
 ------------
 
+Files
+-----
+
+* On startup, the list of banned hosts and networks (via `setban` RPC) in
+  `banlist.dat` is ignored. Bitcoin Core version 22.x is the only version that
```
Comment (jonatack) [inline doc/release-notes.md:73]: (if I'm understanding your intention here)
```suggestion
  `banlist.dat` is ignored. Bitcoin Core version 22.x is the last version that
```

Comment (flack) [reply doc/release-notes.md:73]: it's the last version that can read .dat and the only version that can read .dat _and_ write to .json, so you're both right :-) Maybe just split it into two separate clauses?

**Review by vasild — APPROVED** [review]
ACK fae540b70ae41db08a5999c621fdd896dca8d423

**Comment by flack** [comment]
Does this need some more generic notice for upgrades from older versions? I.e. if you upgrade from .21 or earlier directly to 23.x, you will lose your banlist data, unless you upgrade to 22.x first and let it convert it, right?

**Comment by laanwj** [comment]
>  if you upgrade from .21 or earlier directly to 23.x, you will lose your banlist data, 

As I understand it you won't actually *lose* the data, it will just become inaccessible, but the `.dat` will still be there. Which is good. Someone who notices it later could ostensibly download `0.21.x` and run it once to convert the file.

**Comment by flack** [comment]
yes, but by the time they notice, a new json might already have been created with some new entries, and then you'd have to merge it somehow (or lose the new entries). Not a huge deal I guess, but it couldn't hurt to spell out that scenario in the release notes. At least then you'll have something to point to if someone opens a bug report about how their banlist is gone :-).

**Comment by MarcoFalke** [comment]
Pushed a more verbose version

**Review by vasild — APPROVED** [review]
ACK fa7a93ada2df8d88c9cd411b441110ae6ddec2c0

File: doc/release-notes.md [hunk 2]
```diff
@@ -66,6 +66,16 @@ New RPCs
 Build System
 ------------
 
+Files
+-----
+
+* On startup, the list of banned hosts and networks (via `setban` RPC) in
+  `banlist.dat` is ignored and only `banlist.json` is considered. Bitcoin Core
+  version 22.x is the only version that can read `banlist.dat` and also write
+  it in `banlist.json`. If `banlist.json` already exsits, version 22.x will not
```
Comment (flack) [inline doc/release-notes.md:75]: exsits => exists

and maybe "write it to" instead of "write it in" ?

Comment (MarcoFalke) [reply doc/release-notes.md]: Done, thx

**Review by vasild — APPROVED** [review]
ACK fa2c868886a52352631d32c398d98a2740639e86

**Comment by tryphe** [comment]
ACK fa2c868886a52352631d32c398d98a2740639e86

**Review by Zero-1729 — APPROVED** [review]
ACK fa2c868886a52352631d32c398d98a2740639e86

File: doc/release-notes.md [hunk 3]
```diff
@@ -66,6 +66,16 @@ New RPCs
 Build System
 ------------
 
+Files
+-----
+
+* On startup, the list of banned hosts and networks (via `setban` RPC) in
+  `banlist.dat` is ignored and only `banlist.json` is considered. Bitcoin Core
```
Comment (jonatack) [inline doc/release-notes.md:73]: suggestions:

- "(via the `setban` RPC or the GUI peers tab)"
-  "is now ignored and `banlist.json` is read instead.`
- s/double check/verify/

Comment (jonatack) [reply doc/release-notes.md:73]: Hm, the second sentence seems to contradict the first one. Maybe start with the explaining that if banlist.json is not found, it is created from the data in banlist.dat.

Comment (MarcoFalke) [reply doc/release-notes.md:73]: might do if I have to retouch. Otherwise this can be edited in the wiki in about 6 months

**Pull request merged**